### PR TITLE
Allow to set a custom WorkflowOptions when creating a workflow stub

### LINKED ISSUE #57: Worfkow Configuration
allow to set a custom WorkflowOptions when creating a stub

### PR DESCRIPTION
**Pull request by rmanibus**
fix #57

### CODE REVIEW

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java [hunk 1]
```diff
@@ -17,24 +18,54 @@
...
+    /**
+     * Letting this blank will only work if the workflow is only bound to a single worker.
+     * The default value does not necessary bind to the unnamed worker !
+     */
     String worker() default DEFAULT_WORKER;
 
     @Nonbinding
     String workflowId() default DEFAULT_WORKFLOW_ID;
 
     class Literal extends AnnotationLiteral<TemporalWorkflowStub> implements TemporalWorkflowStub {
```
Comment (rmanibus) [inline extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:41]: should we provide a builder for this ?

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java [hunk 2]
```diff
@@ -17,24 +18,54 @@
...
+
     String DEFAULT_WORKER = "io.quarkiverse.temporal.TemporalWorkflowStub.worker.DEFAULT";
 
     String DEFAULT_WORKFLOW_ID = "io.quarkiverse.temporal.TemporalWorkflowStub.workflowId.DEFAULT";
 
+    /**
+     * This allows to customize the workflow stub configuration.
+     * Letting this blank will bind to the unnamed configuration group.
+     */
+    String group() default DEFAULT_WORKFLOW_GROUP_NAME;
```
Comment (rmanibus) [inline extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:30]: is this the right name ? not sure it is explicit enough

File: extension/runtime/src/main/java/io/quarkiverse/temporal/config/WorkflowRuntimeConfig.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.configuration.DurationConverter;
+import io.smallrye.config.WithConverter;
+import io.smallrye.config.WithDefault;
+import io.temporal.api.enums.v1.WorkflowIdConflictPolicy;
+import io.temporal.api.enums.v1.WorkflowIdReusePolicy;
+
+@ConfigGroup
+public interface WorkflowRuntimeConfig {
```
Comment (rmanibus) [inline extension/runtime/src/main/java/io/quarkiverse/temporal/config/WorkflowRuntimeConfig.java:13]: need to provide missing default or make fields optional

File: extension/runtime/src/main/java/io/quarkiverse/temporal/config/RetryRuntimeConfig.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import io.quarkus.runtime.configuration.DurationConverter;
+import io.smallrye.config.WithConverter;
+import io.smallrye.config.WithDefault;
+
+@ConfigGroup
+public interface RetryRuntimeConfig {
+    /**
+     * List of application failures types to not retry
+     */
+    List<String> doNotRetry();
```
Comment (rmanibus) [inline extension/runtime/src/main/java/io/quarkiverse/temporal/config/RetryRuntimeConfig.java:17]: is there a way to default to an empty list ?

**Comment by rmanibus** [comment]
Also: This PR assume everything should be config based, do we think that it can be beneficial to configure this using code ?
For now the only option configurable using business logic is the workflow id.
It is always possible to inject the client and create a stub programmatically.

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java (hunk 2 above)
Comment (melloware) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:30]: i feel like group is OK. Not sure what else you would call it?

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java (hunk 1 above)
Comment (melloware) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:41]: Sure if you want 😄

File: extension/runtime/src/main/java/io/quarkiverse/temporal/config/RetryRuntimeConfig.java (hunk 4 above)
Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/config/RetryRuntimeConfig.java:18]: still failing with @WithDefault(""), @WithDefault("[]") did work

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java (hunk 2 above)
Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:30]: yeah I don't really have a better idea

File: extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java (hunk 1 above)
Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/TemporalWorkflowStub.java:41]: done !

**Pull request merged**